apple/internal/local_provisioning_profiles.bzl

```diff
@@ -0,0 +1,142 @@
+"""# Rules for using locally installed provisioning profiles"""
+
+def _local_provisioning_profiles(repository_ctx):
+    system_profiles_path = "{}/Library/MobileDevice/Provisioning Profiles".format(repository_ctx.os.environ["HOME"])
+    repository_ctx.execute(["mkdir", "-p", system_profiles_path])
+    repository_ctx.symlink(system_profiles_path, "profiles")
+
+    repository_ctx.file("BUILD", """
+filegroup(
+    name = "profiles",
+    srcs = glob(["profiles/*.mobileprovision"], allow_empty = True),
+    visibility = ["//visibility:public"],
+)
+
+filegroup(
+    name = "empty",
+    srcs = [],
+    visibility = ["//visibility:public"],
+)
+
+alias(
+    name = "fallback_profiles",
+    actual = "{}",
+    visibility = ["//visibility:public"],
+)
+""".format(repository_ctx.attr.fallback_profiles or ":empty"))
+
+local_provisioning_profiles = repository_rule(
+    environ = ["HOME"],
+    implementation = _local_provisioning_profiles,
+    attrs = {
+        "fallback_profiles": attr.label(
+            allow_files = [".mobileprovision"],
+        ),
+    },
+    doc = """
+This rule declares an external repository for discovering locally installed
+provisioning profiles. This is consumed by `ios_local_provisioning_profile`.
+You can optionally set 'fallback_profiles' to point at a stable location of
+profiles if a newer version of the desired profile does not exist on the local
+machine. This is useful for checking in the current version of the profile, but
+not having to update it every time a new device or certificate is added.
+
+## Example
+
+load("@build_bazel_rules_apple//apple:apple.bzl", "local_provisioning_profiles")
+
+local_provisioning_profiles(
```

Comment:
these rule names are maybe a bit too similar, maybe we should name this `provisioning_profile_repository` or something? up for suggestions!

Reply:
I think that would be better, and maybe add an example in the docs about how it interacts with `local_provisioning_profile`?

Reply:
Updated!